pkg/metrics/pushclient/pushclient.go

```diff
@@ -0,0 +1,42 @@
+package pushclient
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// PushClient is used to send Prometheus metrics objects to any Prometheus
+// Push Gateway. It stores the URL and the client that will be used to push
+// to the desired gateway.
+type PushClient struct {
+	URL    string
+	Client *http.Client
+}
+
+// pushAllToAggregationGateway is a helper that takes care of the actual code that pushes to the prometheus
+// aggregation gateway. It takes a list of collectors and pushes all of them to the desired url.
+func (p *PushClient) pushAllToAggregationGateway(collectors ...prometheus.Collector) error {
+	allJobsName := "installer_metrics"
+	pushClient := push.New(p.URL, allJobsName).Client(p.Client).Format(expfmt.FmtText)
+
+	for _, value := range collectors {
+		pushClient.Collector(value)
+	}
+
+	err := pushClient.Push()
+	if err != nil {
+		return errors.Wrap(err, "failed to push metrics")
+	}
+	return nil
+}
+
+// Push calls the pushAllToAggregationGateway for the all metrics.
+// This function acts as a re-router which currently pushes to the aggregation gateway.
+// Any change in gateway logic can be written here to route the metrics.
+func (p *PushClient) Push(collectors ...prometheus.Collector) error {
+	return p.pushAllToAggregationGateway(collectors...)
+}
```

Comment:
With the recent changes, this no longer adds any value. So we can drop this and use/export `pushAll...` instead

Reply:
Was thinking if the gateway changes, it would be a simple change here, rather than changing the other function.

Reply:
> Was thinking if the gateway changes, it would be a simple change here, rather than changing the other function.

How can you change the gateway here? This function just calls `pushAllToAggregationGateway` so why not just call that directly?

Reply:
If we just create another function and have this function call that one, I thought it would be less invasive to the workflow. All changes would be local in this file.